src/lib/index.js

```diff
@@ -36,25 +36,11 @@ export function tooLight(color) {
   return hslColor.l > TOO_LIGHT_FACTOR;
 }
 
-function getPlotIcons(plotly) {
-  const icons = Object.keys(plotly.PlotSchema.get().traces).reduce(
-    (allTraces, trace) => {
-      const componentName = `Plot${capitalize(trace)}Icon`;
-      const iconComponent = PlotlyIcons[componentName];
-      if (componentName) {
-        allTraces[trace] = iconComponent;
-      } else {
-        allTraces[trace] = PlotlyIcons.PlotLineIcon;
-      }
-      return allTraces;
-    },
-    {}
-  );
-
-  // We have to add some of the editor specific trace types as they're not
-  // in plotly.PlotSchema.get()
-  icons.line = PlotlyIcons.PlotLineIcon;
-  return icons;
+function renderTraceIcon(trace) {
+  const componentName = `Plot${capitalize(trace)}Icon`;
+  return PlotlyIcons[componentName]
+    ? PlotlyIcons[componentName]
+    : PlotlyIcons.PlotLineIcon;
```

Comment:
okay @bpostlethwaite @VeraZab how does this look?

Reply:
but aren't there some folds that are rendered and don't belong to a trace? so then the icon should just be null?

Reply:
but yeah, otherwise, looks good! that's the idea!

Reply:
yeah so this function should only be called on the traceFold area (or anywhere we want to generate a specific trace icon), so we want to render a trace icon but if the naming is wrong or the icon doesn't exist, we will render the default trace icon (which we can change if `PlotLineIcon` isn't what we want as a default. The fold takes a `icon` prop and if the icon is passed it will render, otherwise it wont.